pkg/deploy/generator/resources.go

```diff
@@ -1582,6 +1589,21 @@ func (g *generator) rpVersionStorageAccount() []*arm.Resource {
 	}
 }
 
+func (g *generator) rpBootDiagnosticsStorageAccount() *arm.Resource {
```

Comment:
1) I think we need a storage account per region.  We don't want VMs in australiasoutheast to be dependent on a storage account somewhere in the US, for example.
2) Look at using [uniqueString()](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#uniquestring) to generate the storage account names, based on a baseString which we store in the [deployment config](https://github.com/jim-minter/ARO-RP/blob/master/pkg/deploy/config.go#L29).

Reply:
> I think we need a storage account per region.

Good point. I completely missed the implication of `global` in the caller method and JSON output file. Will fix.

> Look at using uniqueString() to generate the storage account names, based on a baseString which we store in the deployment config.

I am not sure I understand how `uniqueString()` works with our deployment config. The example in the doc uses the `baseString` to limit the scope of the uniqueness. The returned value is fixed to 13 characters long and it isn't globally unique. We will need to add a prefix to the string to ensure uniqueness. Will something ilke this work:

```json
  "variables": {
    "storageAccountName": "[concat(
          toLower(parameters('rpBootDiagnosticsStorageAccountPrefix')), 
          uniqueString(resourceGroup().id))
     ]"
  },
```

Reply:
For discussion sake, since `uniqueString` can't guarantee globally unique value, do we even need it? Feels like we just need to concatenate a configurable prefix with some static string (e.g. region) for each storage in each region.

Reply:
Per conversation with @jim-minter, we will not use the `uniqueString()` function as it can't guarantee global uniqueness. We will declare the storage account name as a configurable variable in the RP config YAML.